src/pyodide/internal/python.js

```diff
@@ -442,8 +449,13 @@ function encodeSnapshot(heap, dsoJSON) {
  */
 function decodeSnapshot(memorySnapshot) {
   const uint32View = new Uint32Array(memorySnapshot);
-  const snapshotOffset = uint32View[0];
-  const jsonLength = uint32View[1];
+  if (uint32View[0] !== SNAPSHOT_MAGIC) {
+    throw new Error("Memory snapshot doesn't start with snapshot magic number");
```

Comment:
Do we want to have a fallback here to make the existing snapshots work?
@kflansburg @dom96

Reply:
yeah, I'd say so